src/opm/input/eclipse/Schedule/MSW/WellSegments.cpp

```diff
@@ -310,30 +387,47 @@ namespace Opm {
             const double depth_inc = (depth_last - depth_outlet) / number_segments;
             const double volume_segment = m_segments[range_end].crossArea() * length_inc;
 
+            const auto x_outlet = m_segments[outlet_index].node_X();
+            const auto y_outlet = m_segments[outlet_index].node_Y();
+
+            const auto dx = (m_segments[range_end].node_X() - x_outlet) / number_segments;
+            const auto dy = (m_segments[range_end].node_Y() - y_outlet) / number_segments;
+
             for (std::size_t k = range_begin; k <= range_end; ++k) {
                 const auto& old_segment = this->m_segments[k];
-                double new_volume, new_length, new_depth;
+
+                double new_length, new_depth, new_x, new_y;
                 if (k == range_end) {
                     new_length = old_segment.totalLength();
-                    new_depth = old_segment.depth();
-                } else {
-                    new_length = length_outlet + (k - range_begin + 1) * length_inc;
-                    new_depth  = depth_outlet + (k - range_end + 1) * depth_inc;
+                    new_depth  = old_segment.depth();
+                    new_x      = old_segment.node_X();
+                    new_y      = old_segment.node_Y();
                 }
+                else {
+                    const auto numInc = k - range_begin + 1;
```

Comment:
What is the reason for this to be in CamelCase and the other new variables elsewhere not?

Reply:
> What is the reason for this to be in CamelCase and the other new variables elsewhere not?

No reason, just muscle memory.  I've switched to match surrounding style.